### ISSUE TITLE
A-10C-HSI: red flags for From/To and Course not available

### ISSUE THREAD

**Issue by Bigopus** [HeliosVirtualCockpit/Helios#410]
There is no binding in the interface part which corresponds to the HSI flags.

**Comment by BlueFinBima** [comment]
Notes:  HSI Miles Flag is arg 32 and the HSI Course Flag is arg 713

**Comment by Todd1215** [comment]
The OP reference is that the A10C interface does not provide binding values for  HSI To/From flags which is correct. The question is does DCS export these values?

**Comment by Bigopus** [comment]
In original DCS-screen the flags are visible and working. In DCS-Bios you can get the export:

 

 

void onHsiRangeFlagChange(unsigned int newValue) {
    /* your code here */
}
DcsBios::IntegerBuffer hsiRangeFlagBuffer(0x1048, 0xffff, 0, onHsiRangeFlagChange);

 

 

void onHsiTofrom1Change(unsigned int newValue) {
    /* your code here */
}
DcsBios::IntegerBuffer hsiTofrom1Buffer(0x1064, 0xffff, 0, onHsiTofrom1Change);

 

 

void onHsiTofrom2Change(unsigned int newValue) {
    /* your code here */
}
DcsBios::IntegerBuffer hsiTofrom2Buffer(0x1066, 0xffff, 0, onHsiTofrom2Change);

 

Hope that helps!

 

Michael

 

 

Von: wheelchock ***@***.*** 
Gesendet: Mittwoch, 14. April 2021 22:38
An: HeliosVirtualCockpit/Helios ***@***.***>
Cc: Bigopus ***@***.***>; Author ***@***.***>
Betreff: Re: [HeliosVirtualCockpit/Helios] A-10C-HSI: red flags for From/To and Course not available (#410)

 

The OP reference is that the A10C interface does not provide binding values for HSI To/From flags which is correct. The question is does DCS export these values?

**Comment by BlueFinBima** [comment]
@wheelchock I have not verified, but I don't think there is any reason to
believe that the two arguments I mentioned above cannot be exported, and
should be implemented in the interface. Ideally they should also be used in
the HSI supplied.
This said, I have not verified that they are missing, but I suspect they
are.

**Comment by derammo** [comment]
HSI / Range Flag = (0, 32) is already in the Helios interface and also in the DCS-BIOS interface at the same number

Course Flag is not in the Helios interface, at least not with code 713.    

The two other values that @Bigopus references above (I translated from the addresses back to the specs) in DCS-BIOS are the "To/From 1"
 and 
"To/From 2" 
gauge position values (not flags), which are ID 42 and 43.   

These are missing from Helios.

Here is the spec if you want to add them:

```{
    "DCS-BIOS": {
        "device": "HSI",
        "id": 42,
        "key": "HSI_TOFROM1",
        "mappedRange": [
            0,
            1
        ],
        "name": "HSI TO/FROM 1",
        "type": "float",
        "doc": {
            "control_type": "analog_gauge",
            "description": "HSI TO/FROM 1"
        }
    }
{
    "DCS-BIOS": {
        "device": "HSI",
        "id": 43,
        "key": "HSI_TOFROM2",
        "mappedRange": [
            0,
            1
        ],
        "name": "HSI TO/FROM 2",
        "type": "float",
        "doc": {
            "control_type": "analog_gauge",
            "description": "HSI TO/FROM 2"
            ]
        }
    }
    ````

**Comment by derammo** [comment]
@Bigopus there are two separate questions for you:

1) is this the Range flag you are looking for?  It is already in the interface:

![image](https://user-images.githubusercontent.com/817160/114791458-98204f00-9d54-11eb-9eae-5ddfe8f1b66a.png)

2) please confirm the other two values you are looking for are floating point numbers related to two needles/gauges called "To/From 1" and "To/From 2"?

**Comment by Bigopus** [comment]
1. Okay, I confused the bearing flag, it is definitely there. 

2. So only the flags From and To are missing. They show whether I am approaching or departing a VOR or TACAN station. So quite an important navigation display. 
These are flags that are either on or off,  so 1/0.  
You can find a good simulation here: http://www.luizmonteiro.com/learning_vor_sim.aspx

cu Michael

@Bigopus <https://github.com/Bigopus>  there are two separate questions for you:

1.	is this the Range flag you are looking for? It is already in the interface:

 <https://user-images.githubusercontent.com/817160/114791458-98204f00-9d54-11eb-9eae-5ddfe8f1b66a.png> 

2.	please confirm the other two values you are looking for are floating point numbers related to two needles/gauges called "To/From 1" and "To/From 2"?

**Comment by Todd1215** [comment]
It seems that the HSI FROM Flag is 42 "HSI_TOFROM1" and the HSI TO Flag is 43 "HSI_TOFROM2".

My test consisted of a simple flight approaching a dialed in TACAN where the TO flag is in an one state and the FROM flag is in an off state and when moving away from that same TACAN station the TO flag is in an off state and the FROM flag is in a on state.

Oddly I don't see a TO/FROM flag indicated within the in-game A-10 HSI itself.

**Comment by Bigopus** [comment]
The behavior of the flags is correct. Maybe it's just the little arrow on the tail of the airplane symbol on HSI. It appears when you fly directly towards or away from the tacan. Is a different kind of representation

 

Von: wheelchock ***@***.*** 
Gesendet: Donnerstag, 15. April 2021 17:14
An: HeliosVirtualCockpit/Helios ***@***.***>
Cc: Bigopus ***@***.***>; Mention ***@***.***>
Betreff: Re: [HeliosVirtualCockpit/Helios] A-10C-HSI: red flags for From/To and Course not available (#410)

 

It seems that the HSI FROM Flag is 42 "HSI_TOFROM1" and the HSI TO Flag is 43 "HSI_TOFROM2".

My test consisted of a simple flight approaching a dialed in TACAN where the TO flag is in an one state and the FROM flag is in an off state and when moving away from that same TACAN station the TO flag is in an off state and the FROM flag is in a on state.

Oddly I don't see a TO/FROM flag indicated within the in-game A-10 HSI itself.
